### PR TITLE
Support authentication using an alternative gce token and quota attribution to a different project

### PR DESCRIPTION
**Pull request by yungdarek122**

### CODE REVIEW

File: pkg/export/setup/setup.go [hunk 1]
```diff
@@ -129,6 +129,15 @@ func FromFlags(a *kingpin.Application, userAgent string) func(log.Logger, promet
 	a.Flag("export.debug.batch-size", "Maximum number of points to send in one batch to the GCM API.").
 		Default(strconv.Itoa(export.BatchSizeMax)).UintVar(&opts.BatchSize)
 
+	a.Flag("export.token-url", "The request URL to generate token that's needed to ingest metrics to an alternative project").
+		StringVar(&opts.TokenURL)
+
+	a.Flag("export.token-body", "The request Body to generate token that's needed to ingest metrics to an alternative project.").
```
Comment (fabxc) [inline pkg/export/setup/setup.go:135]: request body

Can you elaborate on the "alternative project" bit?

File: pkg/export/gce_token_source.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	"context"
+	"encoding/json"
+	"net/http"
+	"strings"
+	"time"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/googleapi"
+	"k8s.io/client-go/util/flowcontrol"
```
Comment (fabxc) [inline pkg/export/gce_token_source.go:28]: Kubernetes dependencies usually explode and basically always require Go modules hackery to get things to compile. The fewer distinct places we introduce them, the better the odds that we find a constellation that actually compiles.

Is this dependency needed or can we find a good stand-alone alternative?

Comment (yungdarek122) [reply pkg/export/gce_token_source.go]: Removed this dependency by using the limiter from golang.org/x/time/rate instead

File: pkg/export/setup/setup.go (hunk 1 above)
Comment (yungdarek122) [reply pkg/export/setup/setup.go]: This bit is incorrect. Updated the text to "to generate token that's needed to ingest metrics to the project".

**Review by andysim3d — APPROVED** [review]
LGTM

**Pull request merged**